src/com/owncloud/android/files/services/TransferRequester.java

```diff
@@ -0,0 +1,317 @@
+ */
+/**
+ * Facade class providing methods to ease requesting commands to transfer services {@link FileUploader} and
+ * {@link FileDownloader}.
+ *
+ * Protects client objects from the verbosity of {@link android.content.Intent}s.
+ *
+ * TODO add methods for {@link FileDownloader}, right now it's just about uploads
+ */
+
+public class TransferRequester {
+
+    private static final String TAG = TransferRequester.class.getName();
+
+    /**
+     * Call to upload several new files
+     */
+    public void uploadNewFile(
+        Context context,
+        Account account,
+        String[] localPaths,
+        String[] remotePaths,
+        String[] mimeTypes,
+        Integer behaviour,
+        Boolean createRemoteFolder,
+        int createdBy
+    ) {
+        Intent intent = new Intent(context, FileUploader.class);
+
+        intent.putExtra(FileUploader.KEY_ACCOUNT, account);
+        intent.putExtra(FileUploader.KEY_LOCAL_FILE, localPaths);
+        intent.putExtra(FileUploader.KEY_REMOTE_FILE, remotePaths);
+        intent.putExtra(FileUploader.KEY_MIME_TYPE, mimeTypes);
+        intent.putExtra(FileUploader.KEY_LOCAL_BEHAVIOUR, behaviour);
+        intent.putExtra(FileUploader.KEY_CREATE_REMOTE_FOLDER, createRemoteFolder);
+        intent.putExtra(FileUploader.KEY_CREATED_BY, createdBy);
+
+        context.startService(intent);
+    }
+
+    /**
+     * Call to upload a new single file
+     */
+    public void uploadNewFile(Context context, Account account, String localPath, String remotePath, int
+        behaviour, String mimeType, boolean createRemoteFile, int createdBy) {
+
+        uploadNewFile(
+            context,
+            account,
+            new String[]{localPath},
+            new String[]{remotePath},
+            new String[]{mimeType},
+            behaviour,
+            createRemoteFile,
+            createdBy
+        );
+    }
+
+    /**
+     * Call to update multiple files already uploaded
+     */
+    private void uploadUpdate(Context context, Account account, OCFile[] existingFiles, Integer behaviour,
+                              Boolean forceOverwrite) {
+        Intent intent = new Intent(context, FileUploader.class);
+
+        intent.putExtra(FileUploader.KEY_ACCOUNT, account);
+        intent.putExtra(FileUploader.KEY_FILE, existingFiles);
+        intent.putExtra(FileUploader.KEY_LOCAL_BEHAVIOUR, behaviour);
+        intent.putExtra(FileUploader.KEY_FORCE_OVERWRITE, forceOverwrite);
+
+        context.startService(intent);
+    }
+
+    /**
+     * Call to update a dingle file already uploaded
+     */
+    public void uploadUpdate(Context context, Account account, OCFile existingFile, Integer behaviour,
+                             Boolean forceOverwrite) {
+
+        uploadUpdate(context, account, new OCFile[]{existingFile}, behaviour, forceOverwrite);
+    }
+
+
+    /**
+     * Call to retry upload identified by remotePath
+     */
+    public void retry (Context context, OCUpload upload) {
```

Comment:
Rename as retryUpload? I know that we are not retrying the downloads currently, but it could be a future feature, and we would have to add here a retryDownload method. :wink: